### DISCUSSION TITLE
Generate New UIDs

### DISCUSSION THREAD

**Discussion by GHNJ** [fo-dicom/fo-dicom#1229] [Q&A] [answered]
Hi, I'm fairly new to fo-dicom, and I'm trying to learn based on the examples, and I'm trying to understand how I'd call the method to actually generate a new SIUID and then apply it to the file in question, and save it. I know how to open the file and then "add/update" a specific tag, but how do I call the "GenerateNew" method so obtain a new UID? 

thank you!!

**Comment by mrbean-bremen** [comment, marked as answer]
`GenerateNew` is deprecated in fo-dicom 4 (it was not always unique) and removed in fo-dicom 5, use `DicomUID.GenerateDerivedFromUUID()` instead:
 ```cs
dataset.AddOrUpdate(DicomTag.SOPInstanceUID, DicomUID.GenerateDerivedFromUUID());
```

**Reply by GHNJ** [reply]
thank you!
